invokeai/backend/flux/modules/autoencoder.py

```diff
@@ -312,12 +312,25 @@ def encode(self, x: Tensor, sample: bool = True, generator: torch.Generator | No
             Tensor: Encoded latent tensor. Shape: (batch_size, z_channels, latent_height, latent_width).
         """
 
+        # VAE is broken in float16, use same logic in model loading to pick bfloat16 or float32
+        if x.dtype == torch.float16:
+            try:
+                x = x.to(torch.bfloat16)
+            except TypeError:
+                x = x.to(torch.float32)
```

Comment:
The `AutoEncoder` class should not be responsible for casting input dtypes - it should simply raise if an incompatible dtype is passed, which it already does.

The casting should happen in the caller (`flux_vae_encode.py` / `flux_vae_decode.py`). And rather than mirroring the model loading logic, we should simply cast the input to match the dtype of the model.

Reply:
How do I get the dtype of the model, I use the same check as the loading I couldn't find a simple way to do that.

Reply:
Yep, checking the type of the first param is reasonable. I'd recommend doing it like this instead though: `next(iter(vae.parameters())).dtype`. IIRC, `.parameters()` is a generator while `.state_dict()` constructs the full state_dict, so the former should be slightly faster.